fix(common): type conversion from str for a custom resource

iulianbarbu opened the pull request:
## Description of change

We're missing the `Custom` resource type conversion from `str`.

## How has this been tested? (if applicable)

Reproduced when I deployed the `custom-resource/pdo` example.
TODO: test the project deployment again after the fix.

oddgrd reviewed (approved):
This gives me a feeling of déjà vu, I thought we fixed this a while back :sweat_smile:

iulianbarbu commented:
I was surprised to see the error too. Not sure what we've fixed the last time, or if we introduced a regression in the meantime. I haven't digged into it, but it might not be worth doing it.
